tests/src/test/scala/org/http4s/multipart/MultipartParserSpec.scala

```diff
@@ -124,10 +156,54 @@ object MultipartParserSpec extends Specification {
         |here's another test
         |catch me if you can!
         |
+        |--_5PHqf8_Pl1FCzBuT5o_mVZg36k67UYI--""".stripMargin
+      val input = ruinDelims(unprocessedInput)
+
+      val results: Process0[Headers \/ ByteVector] =
+        unspool(input, 15) pipe MultipartParser.parse(boundary, 100)
+
+      results.toVector must throwA(ParseFailure("Invalid multipart entity", "Part header was longer than 100-byte limit"))
+    }
+
+    "handle an miserably large body on one line" in {
+      val input = ruinDelims("""--_5PHqf8_Pl1FCzBuT5o_mVZg36k67UYI
+        |Content-Disposition: form-data; name="upload"; filename="integration.txt"
+        |Content-Type: application/octet-stream
+        |Content-Transfer-Encoding: binary
+        |
+        """.stripMargin)
+      val end = "--_5PHqf8_Pl1FCzBuT5o_mVZg36k67UYI--"
+
+      val expectedHeaders = Headers(
+        `Content-Disposition`("form-data", Map("name" -> "upload", "filename" -> "integration.txt")),
+        `Content-Type`(MediaType.`application/octet-stream`),
+        Header("Content-Transfer-Encoding", "binary")
+      )
+
+      val body = emit(ByteVector.view("Misery is the river of the world".getBytes("ASCII"))).repeat.take(100000)
```

Comment:
A good test increases developer confidence, guards against regressions, and introduces the reader to the music of Tom Waits.